src/Native/ObjWriter/llvmCap/CMakeLists.txt

```diff
@@ -0,0 +1,116 @@
+cmake_minimum_required(VERSION 3.6)
+project(ObjWriter)
+include(ExternalProject)
+
+set(OBJWRITER_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../)
+set(OBJWRITER_LLVM_POINT tools/ObjWriter)
+
+if(NOT OBJWRITER_TARGET_ARCH)
+  if(CLR_CMAKE_TARGET_ARCH MATCHES "^(x64|x86)$")
+    if(NOT $ENV{__CrossBuild})
```

Comment:
It would be cleaner to pass this as a CMake variable in gen-buildsys-clang.sh rather than using an env var. Basically, you'd just add the variable definition to the cmake_extra_defines in there based on the __CrossBuild value.

Reply:
I thought about it, it seemed to me, it would complicate gen-buildsys-clang.sh logic. But if it's desirable to maintain a single style, I can do a little to change, so that there is no use of external variables from CMake.

Reply:
It is a matter of single style. It seems strange to have some parameters in env vars and some passed as defined variables using the cmake -D option. I personally prefer the -D option since it is more natural to cmake and also makes the behavior of the generated make files static. So basically, you generate make files and then when you run make, it always builds the same way no matter what the env settings are. 
We use cmake in a slightly twisted way, the "native" way as I understand it is to run cmake only if the CMakeLists.txt are changed. Otherwise just run make.

Reply:
Ok, I'll fix.